### ISSUE TITLE
"Long" inter-node transfers on pod network often hang on Vagrant

### ISSUE THREAD

**Issue by bmbroom** [kubernetes/kubernetes#13920]
I am experiencing frequent hangs on large transfers (> 12 KB)  between nodes when using the pod network;  that is, between 10.246.x.n and 10.246.y.m for x != y.   The larger the transfer the more likely the hang.  I first noticed the problem on http responses of 33 kb, which hang at about the 12 kb mark 50% time and succeed the rest of the time.  Larger transfers hang more frequently but always at about the 12 kb mark. I see the problem with both http and ssh (scp) transfers.  I see the problem for all four combinations of (node,pod) on one node to (node,pod) on another.  I see the problem for master to minion and minion to minion transfers.

I do not see any problem within a single node (10.246.x.n to 10.246.x.m).  I do not see any problem when using the host network (10.245 addresses).

I have tried Kubernetes 1.0.5 using Vagrant 1.7.4 and VirtualBox 5.0.2 on a Mac host, as well as Kubernetes 1.0.6 using Vagrant 1.7.4 and VirtualBox 5.0.4 on a Linux host.  Both show the same problem.  I even tried switching the os for the vagrant nodes from Fedora 21 to 22.  That caused lots of other problems, but this network hanging issue was still there.

I looked at the network traffic using Wireshark.  When the problem occurred, an ICMP redirect message would be sent from the pod ip of the host node (10.246.x.1) to the pod ip of the server (10.246.x.4 in my case) redirecting the stream to the host ip of the destination (10.246.y.1) (which was also the destination).  This was immediately followed by a retransmission of the original packet with the same IP addresses (i.e. from 10.246.x.4 to 10.246.y.1) but with different mac addresses.  Wireshark flagged this second packet as a retransmission.

The same sequence of 3 packets would then repeat periodically (except Wireshark would flag both data packets as retransmissions).

**Comment by derekwaynecarr** [comment]
Were you attempting to access the resource from outside the cluster?

For example, were you trying to do something like:

```
https://10.245.1.2/api/v1/proxy/namespaces/kube-system/services/kube-ui
```

There is an issue I am trying to debug that is similar to this where larger content may intermittently hang when proxying.  I am looking to see if moving to Flannel instead of OVS may help resolve the issue.

**Comment by bmbroom** [comment]
Originally I was trying to access a service (10.247 address) from outside the cluster through the host interface.  I eventually tracked it down to accessing the service from a different node, and then to just transfers within the 10.246 network.  The description I gave above is purely within the 10.246 network, but I think it's the cause of problems accessing a service from the host through the 10.245 interface.

**Comment by derekwaynecarr** [comment]
@bmbroom  I fixed the issue in PR https://github.com/kubernetes/kubernetes/pull/13986, if you can test it out to verify, it would be much appreciated.  Things appear to work well for me now.  Thanks!

**Comment by bmbroom** [comment]
@derekwaynecarr I pulled that PR and it all appears to work for me now.  Thanks very much!

**Comment by derekwaynecarr** [comment]
@bmbroom thanks for the report!

**Comment by derekwaynecarr** [comment]
See #14052 for when fix will be merged in the next 1.0.x release stream if working from master branch is not sufficient for you.
